src/app/clusters/general-diagnostics-server/general-diagnostics-server.cpp

```diff
@@ -290,6 +323,50 @@ GeneralDiagnosticsDelegate gDiagnosticDelegate;
 
 } // anonymous namespace
 
+bool emberAfGeneralDiagnosticsClusterTestEventTriggerCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                              const Commands::TestEventTrigger::DecodableType & commandData)
+{
+
+    if (commandData.enableKey.size() != TestEventTriggerDelegate::kEnableKeyLength)
+    {
+        commandObj->AddStatus(commandPath, Status::ConstraintError);
+        return true;
+    }
+
+    if (IsByteSpanAllZeros(commandData.enableKey))
+    {
+        commandObj->AddStatus(commandPath, Status::ConstraintError);
+        return true;
+    }
+
+    auto * triggerDelegate = Server::GetInstance().GetTestEventTriggerDelegate();
+
+    if (triggerDelegate == nullptr || !triggerDelegate->DoesEnableKeyMatch(commandData.enableKey))
+    {
+        commandObj->AddStatus(commandPath, Status::UnsupportedAccess);
+        return true;
+    }
+
+    Status returnStatus                 = Status::Success;
+    CHIP_ERROR handleEventTriggerResult = triggerDelegate->HandleEventTrigger(commandData.eventTrigger);
+
+    if (handleEventTriggerResult == CHIP_NO_ERROR)
+    {
+        returnStatus = Status::Success;
```

Comment:
I was thinking something more like an unconditional:
```
  returnStatus = StatusIB(handleEventTriggerResult).mStatus;
```
which lets the callee return whatever status they want...  In particular, that will turn CHIP_NO_ERROR into Status::Success.